File: APIdoc/python/tutorials/listFoldersCreateLink.py
import sys, getopt

email, password, apiKey = sys.argv[1], sys.argv[2], sys.argv[3]

exec(open("../APIDoc.py").read())

sessionKey = auth(email, password, apiKey)

accessRules = list_access_rules(sessionKey)

folders = list_folders(sessionKey, accessRules["RESULT"]["DATA"][2][1])["RESULT"]["DATA"]

for f in folders:
	if(f[2] == "FILE"):
		link = file_create_link(sessionKey, f[16], f[0], f[1], f[4], f[3])
		print(link)

File: APIdoc/ruby/tutorials/listFoldersCreateLink.rb
require_relative '../APIDoc.rb'

email, password, apiKey = ARGV[0], ARGV[1], ARGV[2]

apiDoc = APIDoc.new(email, password, apiKey)

sessionKey = apiDoc.sessionKey()

accessRules = apiDoc.listAccessRules(sessionKey)

folders = apiDoc.listFolders(sessionKey, accessRules[2][1])

for f in folders
	if (f[2] == "FILE")
		link = apiDoc.fileCreateLink(sessionKey, f[16], f[0], f[1], f[4], f[3])
		print(link)
	end		
end